code/game/gamemodes/dynamic/dynamic_rulesets_latejoin.dm

```diff
@@ -0,0 +1,170 @@
+	weight = 7
+	cost = 5
+	requirements = list(40,30,20,10,10,10,10,10,10,10)
+	high_population_requirement = 10
+	repeatable = TRUE
+	flags = TRAITOR_RULESET
+
+/datum/dynamic_ruleset/latejoin/infiltrator/execute()
+	var/mob/M = pick(candidates)
+	assigned += M
+	candidates -= M
+	M.mind.add_antag_datum(new antag_datum())
+	return TRUE
+
+
+//////////////////////////////////////////////
+//                                          //
+//             WIZARD (LATEJOIN)            //
+//                                          //
+//////////////////////////////////////////////
+
+/datum/dynamic_ruleset/latejoin/wizard
+	name = "Latejoin Wizard"
+	antag_datum = /datum/antagonist/wizard
+	antag_flag = ROLE_WIZARD
+	enemy_roles = list("Security Officer","Detective", "Warden", "Head of Security", "Captain")
+	required_enemies = list(2,2,1,1,1,1,1,0,0,0)
+	required_candidates = 1
+	weight = 1
+	cost = 20
+	requirements = list(90,90,70,40,30,20,10,10,10,10)
+	high_population_requirement = 40
+	repeatable = TRUE
+
+/datum/dynamic_ruleset/latejoin/wizard/ready(var/forced = 0)
+	if(GLOB.wizardstart.len == 0)
+		log_admin("Cannot accept Wizard ruleset. Couldn't find any wizard spawn points.")
+		message_admins("Cannot accept Wizard ruleset. Couldn't find any wizard spawn points.")
+		return FALSE
+
+	return ..()
+
+/datum/dynamic_ruleset/latejoin/wizard/execute()
+	var/mob/M = pick(candidates)
+	assigned += M
+	candidates -= M
+	M.mind.add_antag_datum(new antag_datum())
+	M.forceMove(pick(GLOB.wizardstart))
+	return TRUE
+
+
+//////////////////////////////////////////////
+//                                          //
+//               SPACE NINJA                //
+//                                          //
+//////////////////////////////////////////////
+
+/datum/dynamic_ruleset/latejoin/ninja
+	name = "Space Ninja"
+	antag_datum = /datum/antagonist/ninja
+	antag_flag = ROLE_NINJA
+	enemy_roles = list("Security Officer","Detective", "Warden", "Head of Security", "Captain")
+	required_enemies = list(2,2,1,1,1,1,1,0,0,0)
+	required_candidates = 1
+	weight = 4
+	cost = 10
+	requirements = list(90,90,60,20,10,10,10,10,10,10)
+	high_population_requirement = 20
+
+	repeatable = TRUE
+
+/datum/dynamic_ruleset/latejoin/ninja/execute()
+	var/mob/M = pick(candidates)
+	assigned += M
+	candidates -= M
+	M.mind.add_antag_datum(new antag_datum())
+	return TRUE
+
+//////////////////////////////////////////////
+//                                          //
+//       REVOLUTIONARY PROVOCATEUR          //
+//                                          //
+//////////////////////////////////////////////
+
+/datum/dynamic_ruleset/latejoin/provocateur
+	name = "Provocateur"
+	antag_datum = /datum/antagonist/rev/head
+	antag_flag = ROLE_REV_HEAD
+	restricted_roles = list("AI", "Cyborg", "Security Officer", "Warden", "Detective", "Head of Security", "Captain", "Head of Personnel", "Chief Engineer", "Chief Medical Officer", "Research Director", "Internal Affairs Agent")
```

Comment:
Is this supposed to be "Lawyer" and not IAA?

Reply:
Leftover from /vg/, removing